csharp/ql/src/semmle/code/csharp/controlflow/ControlFlowGraph.qll

```diff
@@ -1871,28 +1871,32 @@ module ControlFlow {
           }
         }
 
+        private predicate directlyThrows(ThrowElement te, ThrowCompletion c) {
+          c.getExceptionClass() = te.getThrownExceptionType() and
+          // For stub implementations, there may exist proper implementations that are not seen
+          // during compilation, so we conservatively rule those out
+          not isStub(te)
```

Comment:
I realise this is the previous logic, but doesn't it make sense to have `isStub` at a higher level, for example the callable or the class?